py/packages/genkit/src/genkit/core/types.py

```diff
@@ -80,99 +91,148 @@ class Code(Enum):
 
 
 class CandidateError(BaseModel):
-    model_config = ConfigDict(extra='forbid')
-
+    model_config = ConfigDict(
+        extra='forbid',
+    )
     index: float
     code: Code
-    message: Optional[str] = None
+    message: str | None = None
+
+
+class DataPart(BaseModel):
+    model_config = ConfigDict(
+        extra='forbid',
+    )
+    text: Any | None = None
+    media: Any | None = None
+    toolRequest: Any | None = None
```

Comment:
Ignore if this is still in progress/intentional, but these types look off?

Reply:
Good catch Alex!